test/com/arpnetworking/metrics/portal/scheduling/impl/OneOffScheduleTest.java

```diff
@@ -0,0 +1,69 @@
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import models.internal.scheduling.Schedule;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+
+public class OneOffScheduleTest {
+
+    private static final ZonedDateTime t0 = ZonedDateTime.parse("2018-01-01T00:00:00Z");
+
+    @Test
+    public void testNextRun() {
+        final Schedule schedule = new OneOffSchedule.Builder()
+                .setWhenRun(ZonedDateTime.parse("2019-01-01T00:00:00Z"))
+                .setRunAtAndAfter(ZonedDateTime.parse("2019-01-01T00:00:00Z"))
+                .build();
+
+        // typical progression, from lastRun=null to lastRun>runUntil
+        assertEquals(
+                Optional.of(ZonedDateTime.parse("2019-01-01T00:00:00Z")),
+                schedule.nextRun(Optional.empty())
+        );
+        assertEquals(
+                Optional.empty(),
+                schedule.nextRun(Optional.of(ZonedDateTime.parse("2019-01-01T00:00:00Z")))
+        );
+
+        // lastRun < runAfter: next run should be null
+        assertEquals(
+                Optional.empty(),
+                schedule.nextRun(Optional.of(ZonedDateTime.parse("2018-12-20T12:34:56Z")))
+        );
+
+        // runAfter < lastRun < runUntil: next run should be null
+        assertEquals(
+                Optional.empty(),
+                schedule.nextRun(Optional.of(ZonedDateTime.parse("2019-01-02T12:34:56Z")))
+        );
+
+        // lastRun > runUntil: next run should be null
+        assertEquals(
+                Optional.empty(),
+                schedule.nextRun(Optional.of(ZonedDateTime.parse("9999-01-01T00:00:00Z")))
+        );
```

Comment:
There is also a case where last run is null, but there is no interval between run at and after and run until that meets the period (e.g. atAndAfter = 1:20 PM, until = 1:40 PM, interval = hourly). It'll never run. This should probably go into the run once schedule's javadoc as a degenerate case ;)

Reply:
Good point! I interpreted this as applying to `PeriodicSchedule` -- is there a similar case for `OneOffSchedule` too? I can't think of one.

Reply:
You are correct, there is not for `OneOffSchedule` (at least I can't think of one).